### PR TITLE
Add `test-types` script back

### PR DESCRIPTION
**Pull request by jablko**
I ran into *an* issue with `test-types` which this PR solves by adding [`"types": []`](https://www.typescriptlang.org/tsconfig#types) -> `**/tsconfig.json`. If it was *the* issue with `test-types`, perhaps it can be restored -> the CI?

#1179 temporarily moved `test-types` -> its own `types.yml` workflow, suggesting it be moved back -> `test` once it's running stably:
> This will be a temporary change until we see dtslint running stably, then we can move `test-types` back under the `test` command.

`types.yml` was subsequently eliminated in #1338.

The issue I encountered was dtslint implicitly including and checking every `node_modules/@types` declaration:
```
Error: Errors in typescript@3.5 for external dependencies:
../../../node_modules/@types/prettier/index.d.ts(537,14): error TS2456: Type alias 'Doc' circularly references itself.

    at /home/nottheoilrig/mdx/node_modules/dtslint/bin/index.js:206:19
    at Generator.next (<anonymous>)
    at fulfilled (/home/nottheoilrig/mdx/node_modules/dtslint/bin/index.js:6:58)
```

This PR solves that by adding `"types": []` -> `**/tsconfig.json`:
```Shell
$ for tsconfig in $(git ls-files '**/tsconfig.json'); do
  git show ":$tsconfig" | jq '.compilerOptions.types = []' > "$tsconfig"
done
```

This will still type check our explicit `node_modules/@types` dependencies, however it will [no longer implicitly include every `node_modules/@types` declaration](https://github.com/mdx-js/mdx/pull/1615#discussion_r692615204).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -28,8 +28,8 @@
     "publish-next": "lerna publish --force-publish=\"*\" --pre-dist-tag next --preid next",
     "test-api": "lerna run test-api",
     "test-coverage": "lerna run test-coverage",
-    "test-types": "lerna run test-types",
-    "test": "yarn build && yarn lint && yarn test-coverage"
+    "test-types": "lerna --concurrency 1 run test-types",
```
Comment (jablko) [inline package.json:31]: [Concurrent dtslint invocations](https://github.com/lerna/lerna/blob/main/core/global-options/README.md#--concurrency) were trampling each other installing TypeScript versions ... At least I *think* that's what was happening [here](https://github.com/mdx-js/mdx/pull/1622/checks?check_run_id=3451812571#step:5:373) -- and [here](https://github.com/mdx-js/mdx/issues/1172) ... Perhaps [here](https://github.com/microsoft/dtslint/issues/267) too ...

**Review by ChristianMurphy — APPROVED** [review]
Sounds alright to me.
Thoughts on this approach @JounQin?

**Comment by JounQin** [comment]
I'm not sure, personally I'd prefer something like `Expect` type utility for `types.test.ts`, and run `tsc --noEmit` for type checking for all `.ts` related files.

But for multiple ts versions, `dtslint` maybe better.

However, I don't think targeting multiple ts versions are worth unless we're not going to always follow latest TypeScript.

**Comment by jablko** [comment]
@JounQin I think most of this change applies equally to `tsc --noEmit` and dtslint: Without `"types": []` we'll implicitly include every `node_modules/@types` declaration in either case, I think?

Can we land this, and roll back the `lerna --concurrency 1` option if/when we drop dtslint?

**Review by JounQin — APPROVED** [review]
OK, whether to drop `dtslint` can be discussed later, let's fix current issues first.

**Review by ChristianMurphy — APPROVED** [review]

**Pull request merged**